java/core/src/test/java/com/lancedb/lance/ScannerTest.java

```diff
@@ -271,6 +275,141 @@ void testScanFragments() throws Exception {
     }
   }
 
+  @Test
+  void testDatasetScannerLimit() throws Exception {
+    String datasetPath = tempDir.resolve("dataset_scanner_limit").toString();
+    try (BufferAllocator allocator = new RootAllocator()) {
+      TestUtils.SimpleTestDataset testDataset = new TestUtils.SimpleTestDataset(allocator, datasetPath);
+      testDataset.createEmptyDataset().close();
+      int totalRows = 100;
+      int limit = 50;
+      try (Dataset dataset = testDataset.write(1, totalRows)) {
+        try (Scanner scanner = dataset.newScan(new ScanOptions.Builder().limit(limit).build())) {
+          testDataset.validateScanResults(dataset, scanner, limit, limit);
+        }
+      }
+    }
+  }
+
+  @Test
+  void testDatasetScannerOffset() throws Exception {
```

Comment:
Do we have a unit test for vector search ?

Reply:
Added unit test for vector search
Added the create_index and list_index methods